torch/csrc/jit/fusion_compiler.cpp

```diff
@@ -170,13 +170,15 @@ typedef long long int int64_t;
 ${HalfHeader}
 ${RandHeader}
 #endif
+#define MIN_ONE(N) (N == 0 ? 1 : N)
 typedef ${IndexType} IndexType;
 template<typename T, size_t N>
 struct TensorInfo {
   T * data;
-  IndexType sizes[N];
-  IndexType strides[N];
+  IndexType sizes[MIN_ONE(N)];
+  IndexType strides[MIN_ONE(N)];
 };
```

Comment:
Can we write a template specialization that doesn't have those 2 arrays if `N` is equal to 0? We shouldn't really refer to them in this case, so it might be even easier to catch errors if we do that.